common/src/main/java/com/wynntils/features/overlays/CustomPlayerListFeature.java

```diff
@@ -0,0 +1,240 @@
+
+    @SubscribeEvent
+    public void onRender(RenderEvent.Pre event) {
+        if (event.getType() == RenderEvent.ElementType.PLAYER_TAB_LIST) {
+            event.setCanceled(true);
+        }
+    }
+
+    private static class CustomPlayerListOverlay extends Overlay {
+        private static final Comparator<PlayerInfo> PLAYER_INFO_COMPARATOR =
+                Comparator.comparing(playerInfo -> playerInfo.getProfile().getName(), String::compareToIgnoreCase);
+        private static final int DISTANCE_BETWEEN_CATEGORIES = 87;
+        private static final int ROLL_WIDTH = 27;
+        private static final int HALF_WIDTH = 178;
+        private static final int WIDTH = HALF_WIDTH * 2;
+        private static final int TOTAL_WIDTH = WIDTH + ROLL_WIDTH * 2;
+        private static final int MAX_WIDTH = 73;
+
+        @RegisterConfig
+        public final Config<Integer> openingDuration = new Config<>(125);
+
+        private double animationProgress = 0;
+        private long lastTime = -1;
+
+        protected CustomPlayerListOverlay() {
+            super(
+                    new OverlayPosition(
+                            0,
+                            0,
+                            VerticalAlignment.TOP,
+                            HorizontalAlignment.LEFT,
+                            OverlayPosition.AnchorSection.TOP_MIDDLE),
+                    TOTAL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height());
+        }
+
+        private List<StyledText> lastPlayers = new ArrayList<>();
+        private long nextExecution = 0;
+
+        private List<StyledText> getAvailablePlayers() {
+            if (System.currentTimeMillis() < nextExecution) return lastPlayers;
+
+            nextExecution = System.currentTimeMillis() + 250;
+
+            PlayerTabOverlay defaultTabList = McUtils.mc().gui.getTabList();
+
+            lastPlayers = McUtils.player().connection.getListedOnlinePlayers().stream()
+                    .sorted(PLAYER_INFO_COMPARATOR)
+                    .limit(80)
+                    .map(defaultTabList::getNameForDisplay)
+                    .map(StyledText::fromComponent)
+                    .filter(styledText -> !styledText.contains(ChatFormatting.BOLD.toString()))
+                    .map(StyledText::getString)
+                    .map(styledText -> RenderedStringUtils.cut(styledText, MAX_WIDTH))
+                    .map(styledText ->
+                            styledText.replace(ChatFormatting.GRAY.toString(), ChatFormatting.BLACK.toString()))
+                    .map(StyledText::fromString)
+                    .toList();
+
+            return lastPlayers;
+        }
+
+        @Override
+        public void render(PoseStack poseStack, MultiBufferSource bufferSource, float partialTicks, Window window) {
+            if (!McUtils.options().keyPlayerList.isDown() && animationProgress <= 0.0) return;
```

Comment:
If you render at the correct hook, then you don't need this keybind check.

Reply:
When I render at the correct hook the animation stops working because render doesn't get called after you have released the `tab` key. It would have worked if I had there `||` instead of `&&`